Extended documentation of app charts (#155)
epinio/docs: docs/howtos/create_custom_appcharts.md

If the new Helm chart is not correctly registered in Epinio, check that the
`RELEASE_NAMESPACE` was properly set in `fluentd-appchart.yaml`.

## Going further

> The standard app chart, as well as the chart created here use a kubernetes `Deployment` as
> the main resource describing the active application.

Reviewer: ```suggestion
Both the standard and custom app charts use a kubernetes `Deployment` as
the main resource describing the active application.
```

Author: The new phrase makes it sound as if all custom app charts will use a `Deployment`. In light of the change allowing more than `Deployment` I wanted to call out the tutorial's custom chart specifically, instead saying it generally, as the new phrasing. A slight foreshadowing of the next sentence where the restriction is explained and lifted.

Am I making sense ?

Reviewer: yes, so please discard my comment

> Up to version 1 of Epinio this was the only kind of resource which was acceptable.

Reviewer: ```suggestion
Up to Epinio version 1.0, this was the only kind of resource supported.
```

> After that it becomes possible to use other kinds of controllers, for example `StatefulSet`.

Reviewer: ```suggestion
Since Epinio version 1.0+, other kinds of controllers, for example `StatefulSet`, are supported.
```

> However, even when changing controllers it is important to keep the `Pod` annotations and
> labels the same as for the standard chart, so that Epinio's server can still locate and
> use them properly.

Reviewer: ```suggestion
However, even when changing controllers, it is important to keep the `Pod` annotations and
labels the same as for the standard chart. This allows Epinio's server to locate the required resources and
use them properly.
```

> As examples:

Reviewer: ```suggestion
Here's couple examples:
```

Author: How about

```suggestion
Here's a couple of examples:
```

  1. `epinio.io/stage-id` provides the id of the staging Pod which created the application's image.

>   1. `epinio.io/app-container` provides the name of the application's main `Container`,
>      for use by `epinio exec`, for example. This label has to match the actual name of the
>      container in the pod spec, of course.

Reviewer: ```suggestion
  1. `epinio.io/app-container` provides the name of the application's main `Container`,
     which is used by `epinio exec`. This label has to match the actual name of the
     container in the pod spec.
```
